### PR TITLE
Accessing the API as an integration

### PR DESCRIPTION
**Pull request by naderman**
- Add JWT authentication introduced with GitHub integrations
- Add endpoint for Installation access tokens required for API use from GitHub integrations

### CODE REVIEW

**Comment by naderman** [comment]
Currently I'm using the [locobucci/jwt](https://github.com/lcobucci/jwt/tree/3.2.0) library to create a JSON Web Token. Not sure if I should add this as a dependeny here and provide a convenience method, or if this should just go into the documentation? To actually get an access token for use with the API the following is currently necessary:

```php
        $jwt = (new Builder)
            ->setIssuer($integrationId)
            ->setIssuedAt(time())
            ->setExpiration(time() + 60)
            ->sign(new Sha256(),  (new Keychain)->getPrivateKey($pemPrivateKeyPath))
            ->getToken();

        $github->authenticate($jwt, null, Github\Client::AUTH_JWT);

        $token = $github->api('installations')->createAccessToken($installationId);
        $github->authenticate($token['token'], null, Github\Client::AUTH_HTTP_TOKEN);
```

**Comment by naderman** [comment]
Not sure what the build failures are about.

**Comment by naderman** [comment]
@cursedcoder what are your thoughts on my question above? Otherwise this works as-is, so would be great if you could merge it. There's probably more features one could add later, but this is enough to work with github integrations.

**Comment by cursedcoder** [comment]
@naderman could you please add a doc with this content https://github.com/KnpLabs/php-github-api/pull/439#issuecomment-249991521?

File: lib/Github/Api/Installations.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+     *
+     * @return array token and token metadata
+     */
+    public function createAccessToken($installationId, $userId = null)
+    {
+        $parameters = array();
+        if ($userId) {
+            $paramters['user_id'] = $userId;
+        }
+        return $this->post('/installations/'.rawurlencode($installationId).'/access_tokens', $parameters);
```
Comment (cursedcoder) [inline lib/Github/Api/Installations.php]: should be separated with 1 space line above

Comment (cursedcoder) [inline lib/Github/Api/Installations.php]: https://styleci.io/analyses/8mryaW there's a proper diff how it should look lile

Comment (naderman) [inline lib/Github/Api/Installations.php]: Haha sorry, thought you meant you wanted spaces around concetanation, will add a blank line ;-) Will comment when I finished writing some docs for this.

**Comment by naderman** [comment]
@cursedcoder Alright I added some documentation, and renamed the endpoint to Integrations as per the current API docs. There will likely be a sub object installations at some point. I tested this against the actual API and works for me.

**Comment by cursedcoder** [comment]
thanks !

**Pull request merged**